pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -428,16 +428,34 @@ func (r *ReconcileVirtualMachineImport) createVM(provider provider.Provider, ins
 	targetVMName := mapper.ResolveVMName(instance.Spec.TargetVMName)
 
 	// Define VM spec
+	// Update condition to VM template matching:
+	processingCond := conditions.NewProcessingCondition(string(v2vv1alpha1.VMTemplateMatching), "Matching virtual machine template", corev1.ConditionTrue)
+	if err := r.upsertStatusConditions(instanceNamespacedName, processingCond); err != nil {
+		return "", err
+	}
 	template, err := provider.FindTemplate()
 	var spec *kubevirtv1.VirtualMachine
+	config, cfgErr := r.kvConfigProvider.GetConfig()
+	if cfgErr != nil {
+		log.Error(cfgErr, "Cannot get KubeVirt cluster config.")
+	}
 	if err != nil {
 		reqLogger.Info("No matching template was found for the virtual machine. Using empty VM definition")
+		if !config.ImportWithoutTemplateEnabled() {
+			if err := r.templateMatchingFailed(instanceNamespacedName, err.Error(), &processingCond, provider); err != nil {
+				return "", err
+			}
+			return "", err
+		}
 		spec = mapper.CreateEmptyVM(targetVMName)
 	} else {
 		reqLogger.Info("A template was found for creating the virtual machine", "Template.Name", template.ObjectMeta.Name)
 		spec, err = provider.ProcessTemplate(template, targetVMName, instance.Namespace)
 		if err != nil {
 			reqLogger.Info("Failed to process the template. Using empty VM definition. Error: " + err.Error())
+			if !config.ImportWithoutTemplateEnabled() {
```

Comment:
same as above. Let's first just record the failure of processing the template and after the condition log the message of using empty VM definition.

Reply:
Done